src/peppr/metric.py

```diff
@@ -1557,6 +1583,343 @@ def smaller_is_better(self) -> bool:
+        ligand_is_protein: bool = False,
+        binding_site_cutoff: float = 4.5,
+        number_of_res_in_shard: int = 2,
+        scale_score: bool = False) -> float:
+        score = 0.0
+        if ligand_is_protein:
+            count_df = self._get_interaction_counts_for_ppi(
+                receptor,
+                ligand,
+                binding_site_cutoff=binding_site_cutoff,
+                number_of_res_in_shard=number_of_res_in_shard
+                )
+            log_odd_dict = PPI_LOG_ODDS
+
+        else:
+            count_df = self._get_interaction_counts_for_pli(
+                receptor,
+                ligand,
+                binding_site_cutoff=binding_site_cutoff
+                )
+            log_odd_dict = PLI_LOG_ODDS
+        count_df["resname"] = count_df["residue"].str.split(":").str[1]
+        residue_interaction_tuples = count_df[["resname", "interaction", "count"]].itertuples(index=False)
+        for resname, interaction, count in residue_interaction_tuples:
+            log_odd = log_odd_dict.get(interaction, {})
+            log_odd = log_odd.get(resname, log_odd["XXX"])
+            # Scale score by 10 and multiply by count of interaction
+            score += 10 * log_odd * count
+        if scale_score:
+            # Scale score to be between 0 and 1 using a logistic function
+            upper_asymptote = SCALING_FACTORS["upper_asymptote"]
+            midpoint = SCALING_FACTORS["midpoint"]
+            steepness = SCALING_FACTORS["steepness"]
+            lower_asymptote = SCALING_FACTORS["lower_asymptote"]
+            score = upper_asymptote /(1 + np.exp(-steepness * (score - midpoint))) + lower_asymptote
+        return score
+
+    def evaluate(self, pose_receptor: struc.AtomArray, pose_ligand: struc.AtomArray) -> float:
+        if pose_receptor.array_length() == 0 or pose_ligand.array_length() == 0:
+            return np.nan
+
+        try:
+            return self._calculate_rilo_score(
+                receptor=pose_receptor,
+                ligand=pose_ligand,
+                ligand_is_protein=self._ligand_is_protein,
+                binding_site_cutoff=self._binding_site_cutoff,
+                number_of_res_in_shard=self._number_of_res_in_shard,
+                scale_score=self._scale_score)
+        except struc.BadStructureError:
+            return np.nan
+
+    def smaller_is_better(self) -> bool:
+        return False
+
+
+class iPackingEntropy(Metric):
+    """
+    Interface Packing Entropy (iPackingEntropy) is a metric for evaluating the
+    quality of receptor-ligand interactions.
+    It quantifies how well-packed the interface between the receptor and ligand is,
+    based on the distribution of distances between atoms at the interface.
+
+    The metric is calculated by dividing the interface into small "shards" of residues
+    and evaluating the packing entropy for each shard.
+    The final score is the average packing entropy across all shards.
```

Comment:
I think A formula would help here as well to explain the metric: https://sphinx-rtd-trial.readthedocs.io/en/latest/ext/math.html